Products/ZenUI3/browser/resources/js/zenoss/HierarchyTreePanel.js

```diff
@@ -556,10 +556,10 @@
                     function (success, node) {
                         var lastNode = node;
                         if (success && node) {
-                            node = node.findChild(field, last);
-                            if (node) {
-                                me.getSelectionModel().select(node);
-                                Ext.callback(callback, scope || me, [true, node]);
+                            child = node.findChild(field, last, true);
+                            if (child) {
+                                me.getSelectionModel().select(child);
+                                Ext.callback(callback, scope || me, [true, child]);
```

Comment:
Similar issue here.  This depends on `expandNodeByPath` to traverse down a path and then selects the desired child of the returned node.  Doing a deep search may yield a node that wasn't requested.